### ISSUE TITLE
long runtime for some examples

### ISSUE THREAD

**Issue by JanHasenauer** [genssi-developer/GenSSI#11]
The runtime for some examples is rather long, e.g. for Pharmacokinetics. I think it might be good to warn the user. We could provide the runtime as a comment in the runXXX.m file. For the examples which run really long, we might even want to ask the user to confirm that he/she actually wants to run the example.  

We could used something like:

```
prompt = ['The structural identifiability analysis will take roughly X min. \n' ...
          'Please confirm that you want to run. (yes/no)  '];
if strcmp(input(prompt,'s'),'yes')
    % run code
end
```

What do you think?

**Comment by genssi-developer** [comment]
I think this is a very nice way to warn the user who is running everything in interactive mode and watching the screen. However, there are problems when running long examples. For example, assume that I start a long example or start a script containing all run scripts, and then let them run over lunch or over night. If I set options.store=false, I will have no results, except for the last part of the command window. Storing the results also has the advantage of examining them in an editor, where it is easier to scroll and search. If I set options.closeFigure=false, I will have tons of open figures that I can't correlate with the examples. And, if I use this prompt in the run file, it will be waiting for user input and nothing will have run.
In other words, all three of these things break automation, so I don't like them, unless we have a way of adding code that checks to see if the script is running in the foreground or background.

**Comment by JanHasenauer** [comment]
Hm, sound for me like two different discussion:
1) Defaults for genssiMain
2) Whether or not to ask for the user input

Regarding 1)
Your points regarding automations are obviously correct. In my opinion, this however something which is only relevant for the developers. I think we should however focus on the normal user instead of the developer. For a developer it is anyhow super straight forward to change the defaults in genssiMain before a long runs.
In particular for users which just get started, it is in my opinion much better if they do not first have to go to another folder to reopen the figures and the text files.

Potential solution: The problem could be solved completely if we include a file containing user-specific defaults. This could be generated as part of the installation procedure but ignored using .gitignore. This would allow you to have different default options than me. I will try this.

Regarding 2)
The manual feedback would indeed limit automation substantially. I woudl suggest to use a global default option to check this.

**Comment by JanHasenauer** [comment]
So, I included included genssiInstall. This file includes all necessary parts of GenSSI in the matlabpath and generate the file genssiUserSpecificDefaults.m. genssiUserSpecificDefaults.m provides default options which are used by other routines, including genssiMain. This allows different users to have different defaults. genssiUserSpecificDefaults.m is included in .gitignore. I think this solves the problem. What's your opinion?

Regarding 2)
Given genssiUserSpecificDefaults.m we could also refine the warning for the user, e.g.:

```
function genssiAskForConfirmation(expectedRuntime)

    % Check that installation was performed
    if ~exist('genssiUserSpecificDefaults.m')
        error('Please run genssiInstall.m before performing your first analysis.');
    end

    % Check that installation was performed
    if genssiUserSpecificDefaults('feedbackRequired')
        prompt = ['The structural identifiability analysis will take roughly ' num2str(expectedRuntime) ' min. \n' ...
          'Please confirm that you want to run. (yes/no)  '];
        if ~strcmp(input(prompt,'s'),'yes')
            error('Calculation discontinued by user')
        end
    end
end 
```

genssiAskForConfirmation(expectedRuntime) could be called in the beginning of each example with the respective expected runt time. If a user sets the default for feedbackRequired to false, the whole part is skipped, allowing for automation.

What's your opinion?

**Comment by JanHasenauer** [comment]
I implemented this as an example in the Goodwin model. Please have a look and provide me with feedback.

**Comment by genssi-developer** [comment]
I tried it out and it is working as designed.

**Comment by JanHasenauer** [comment]
Glad to hear. If you think it is a good idea, it would be great of you could implement the check for all models.

**Comment by genssi-developer** [comment]
I have added this call to all run scripts, and tested all examples. However, I deleted the results files before synchronizing to GitHub. I plan to rerun all of them after we understand the problem with "solve" in Pharmacokinetics, and the update the SI document.

**Comment by JanHasenauer** [comment]
Sounds good. Thanks.
